From 8c7f91a5456a594715c963930ed6c6eb9f97db99 Mon Sep 17 00:00:00 2001
From: "hong.wang" <hong.wang@amlogic.com>
Date: Thu, 18 May 2023 10:31:28 +0800
Subject: [PATCH 5/8] BT: fix w400 sink audio sound lag [1/1]

PD#SWPL-119743

Problem:
w400 sink audio sound lag.

Solution:
add bt-hal audio buffer.

Verify:
W400

Change-Id: I37cc9ade1e2bed54b71bbf304a6e3e92ba0a9ed3
Signed-off-by: hong.wang <hong.wang@amlogic.com>
---
 recipes-multimedia/bluez-alsa/bluez-alsa.bb         |  1 +
 .../0001-BT-fix-sink-audio-sound-lag-1-1.patch      | 13 +++++++++++++
 2 files changed, 14 insertions(+)
 create mode 100755 recipes-multimedia/bluez-alsa/files/0001-BT-fix-sink-audio-sound-lag-1-1.patch

diff --git a/recipes-multimedia/bluez-alsa/bluez-alsa.bb b/recipes-multimedia/bluez-alsa/bluez-alsa.bb
index 7a485c26..29f5af3b 100644
--- a/recipes-multimedia/bluez-alsa/bluez-alsa.bb
+++ b/recipes-multimedia/bluez-alsa/bluez-alsa.bb
@@ -15,6 +15,7 @@ SRC_URI += "file://0001-bt-fix-bluealsa-pcm-device-can-t-found-1-2.patch"
 SRC_URI += "file://0001-BT-add-a2dp_ctl-hfp_ctl-1-1.patch"
 SRC_URI += "file://0001-Allow-user-to-enable-profiles-selectively.patch"
 SRC_URI += "file://0001-BT-fix-a2dp_ctl-scan-crash-in-64bit-env-1-1.patch"
+SRC_URI += "file://0001-BT-fix-sink-audio-sound-lag-1-1.patch"
 SRC_URI += "file://bluez-alsa.sh"
 
 
diff --git a/recipes-multimedia/bluez-alsa/files/0001-BT-fix-sink-audio-sound-lag-1-1.patch b/recipes-multimedia/bluez-alsa/files/0001-BT-fix-sink-audio-sound-lag-1-1.patch
new file mode 100755
index 00000000..1a2fe64c
--- /dev/null
+++ b/recipes-multimedia/bluez-alsa/files/0001-BT-fix-sink-audio-sound-lag-1-1.patch
@@ -0,0 +1,13 @@
+Index: git/utils/bt-halplay.c
+===================================================================
+--- git.orig/utils/bt-halplay.c
++++ git/utils/bt-halplay.c
+@@ -224,7 +224,7 @@ static void *pcm_worker_routine(void *ar
+ 		error("output speaker open failed\n");
+ 		goto fail;
+ 	}
+-	snd_pcm_uframes_t period_size = 128;
++	snd_pcm_uframes_t period_size = 128*4;
+ 	ssize_t frame_size = 2 * w->transport.channels; //snd_pcm_frames_to_bytes(w->pcm, 1); //16bit 2channels
+ 	buffer = malloc(period_size * frame_size);
+ 	char *buffer_tail = buffer;
-- 
2.25.1

